avm/res/web/serverfarm/tests/e2e/defaults/main.test.bicep

```diff
@@ -0,0 +1,60 @@
+targetScope = 'subscription'
+
+metadata name = 'Using default parameter set'
+metadata description = 'This instance deploys the module with a base set of parameters. Note it does include the use of Availability zones by default.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-web.serverfarms-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'wsfmin'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(3)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}-${iteration}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: location
+    sku: {
+      name: 'P1v3'
+      tier: 'Premium'
+      size: 'P1v3'
+      family: 'P'
+      capacity: 3
+    }
+    zoneRedundant: true
```

Comment:
Would it not be more meaningful to not set this value to test the default (which is the purpose of this nugget of a test ;) )

Reply:
@AlexanderSehr so it actually was set standard and and no ZR originally! But then it would fail the PSRule tests, which apparently isn't ideal..

Reply:
It's not, right @tsc-buddy. We'd aspire to make the defaults as 'waf-compliant' as possible. Which in turn begs the question, if the logic in the module should be updated to default to ZR. 

I guess I'm a bit suprised. The logic is currently:
```bicep
param zoneRedundant bool = (sku.tier == 'Premium' || sku.tier == 'ElasticPremium') ? true : false
```

And the example sku you provide specifies the `tier` 'Premium'. I'd hence expect that the above logic defaults to `true`. I must be missing something :D

Reply:
Thanks Alex, ill pick this one up tomorrow. Time to clock off...midnight!! :)

Reply:
Have a good one

Reply:
@AlexanderSehr as discussed ive pulled this out of the default test successfully. I went through some old notes from a conversation with Chris and when it was set to standard 1 instance count and logic was on that param at the time, thats when it failed PSRule. The revised test runs fine and no AZ param is set.